RFC.md

```diff
@@ -0,0 +1,393 @@
+
+#### Digest Length in Bytes
+
+The `<digest-length>` counts bytes, not bits. This reduces the storage
+required for most hash function values: 256 and 512 bit lengths are
+represented as "32" and "64", using 1-byte varints instead of 2-byte
+varints.
+
+This size reduction exploits the facts that: (a) most common computer
+architectures use 8-bit words, (b) most networks transit sequences of
+bytes, (c) most storage systems store sequences of bytes, and (d) most
+if not all widely used hash functions have standard digest lengths
+divisible by 8.
+
+It is possible that some hash functions are used with digests of lengths
+not evenly divisble by 8. In such rare cases, the function implementation
+with multihash should define a byte-aligned version, usually by adding a
+pre-determined amount of padding bits at the end of the value. Such
+padding transformations can be well-defined, and would likely have to
+exist to support such hash functions in current computer architectures,
+storage systems, and networks. Therefore this specification is comfortable
+leaving this up to the implementations and users.
+
+#### Digest Value
+
+The `<digest-value-bin>` is simply a binary-packed representation of the
+hash function value. This value ensures the entire Multihash Packed
+Representation is as compact as it could be, wasting no space to
+represent the digest value.
+
+Base encoding for strings must be performed around the whole multihash
+value,
+
+### Explanation Representation
+
+The Explanation Representation is a well defined way of representing
+the hash functions in a human-readable optimized way. It is wasteful
+explicitly, and not meant to be used as an identifier in systems.
+
+The Explanation Representation is
+
+    <hash-function-name>.<digest-length>.<digest-value-hex>
+
+Where:
+- `<hash-function-name>` is the string name of the code of the hash
+  function according to a pre-determined multihash table, and the
+  Standard Multihash Table.
+- `<digest-lengt>` is a decimal number storing the length of
+  `<digest-value>`, in bytes.
+- `<digest-value-hex>` is the hash function digest value, hex-encoded.
+- `.` is a delimiter for the values.
+
+This representation SHOULD NOT be used to transmit, store, or embed a
+multihash value. It SHOULD only be used for debugging.
+
+Even when creating human-oriented string identifiers, it is strongly
+RECOMMENDED to use the packed representation of multihash, possibly
+encoded in a convenient base. This ensures the whole multihash value
+is as compact as it can be, and does not accidentally impose other
+requirements on the transmission of the value. It is much easier for
+systems to deal with a binary value that can be easily encoded in a
+variety of bases. Using the Explanation Representation for storage,
+transmission, embedding in other identifiers, or anything other
+than debugging, defeats the purpose of multihash.
+
+### MSB Unsigned Varints - muvints
```

Comment:
Muvint is a relatively new term -- in most places we've called these unsigned-varint or simply varint.